body.tex

```diff
@@ -0,0 +1,442 @@
+\item be able to activate the correct environment before running and EUPS build using \texttt{eups distrib install}.
+\item be able to run a CI build using a specific environment reference.
+\item be able to manage Conda packages for the third party libraries that requires DM rebuild or customization.
+\end{itemize}
+
+
+\newpage
+\section{Proposed Implementation} \label{sec:proposed}
+
+Here follows the proposed changes required to satisfy the usecases from the previous section.
+
+\subsection{Define Environment References} \label{sec:envDefine}
+
+The package \textit{scipipe\_conda\_env} to be tagged in GitHub for each weekly and each time an EUPS build is created.
+This will guarantee that each EUPS weekly build will have a corresponding well known Conda environment, that can be used for rebuilt them. 
+The Conda environmemt tag shall be in line with the EUPS tag. The SHA1 can be identified also, but should not be strictly needed.
+
+All changes implemented in the \textit{scipipe\_conda\_env} repository shall follow the usual development process.  
+The tooling improvements proposed bellow, will permit the validation of those changes and include them in the development workflow.
+
+
+\subsection{Use Different Environments in lsstsw Git Package} \label{sec:envActivate}
+
+A new command, or command line option, shall be added to lsstsw to deploy and activate a specific Conda environment reference from the \textit{scipipe\_conda\_env} GitHub package.
+
+The possible options should be:
+
+\begin{itemize}
+\item deploy/activate a specific \textit{scipipe\_conda\_env} hash
+\item deploy/activate a specific \textit{scipipe\_conda\_env} branch
+\item deploy/activate a specific \textit{scipipe\_conda\_env} tag
+\item deploy/activate the tip of \textit{scipipe\_conda\_env} master
+\end{itemize}
+
+This can be done with the following changes in lsstsw package.
+
+
+\subsubsection{Update deploy Command} \label{sec:deploy}
+
+The \textbf{deploy} lsstsw command should accept the environment reference \textbf{envref}  as an additional command line option.
+
+The default behavior should be the same as now. If no envref is provided, the usual fixed hash shall be deployed. 
+
+In case the envref is \textit{master}, it shall deploy the latest master, if not already deployed.
+In this way, this option can be used to check if there is a newer master version of the conda environment and deploy it.
+
+The environment shall have a unique name, not the fixed one that is currently used.
+
+
+\subsubsection{Update rebuild Command} \label{sec:rebuild}
+
+The \textbf{rebuild} command should record the environment reference used for each build.
+
+This will permit in the future to retrieve the information, in order to manually activate the same conda environment before running a build.
+
+The \textbf{rebuild} command could also raise a warning, in case the active environment is different from the one previously used.
```

Comment:
You should make clear what "previously used" means and that it may be tricky to determine.

Reply:
I would say: "However this is not a problem related to the environment management but a potential improvement in the build tooling".
If these sentences are confusing... I will remove them. They are partially out of scope here.